source/common/upstream/cluster_manager_impl.cc

```diff
@@ -479,6 +479,37 @@ bool ClusterManagerImpl::addOrUpdateCluster(const envoy::api::v2::Cluster& clust
       // Otherwise, applyUpdates() will fire with a dangling cluster reference.
       updates_map_.erase(cluster_name);
 
+      // If the active cluster has priority sets and the warming cluster does not have them, it
+      // means that onConfigUpdate is triggered for EDS type of cluster is triggered with no
+      // reference to this cluster. In such cases, copy the active cluster priority set to the
+      // warming cluster otherwise the active cluster hosts will be cleared after warming is
```

Comment:
nit: I'd rephrase this to say something like `to prevent the hosts from being cleared after warming`